model/src/main/kotlin/io/spine/validation/CompanionOptions.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2025, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.validation
+
+import com.google.protobuf.GeneratedMessage.GeneratedExtension
+import io.spine.protodata.Compilation
+import io.spine.protodata.ast.Field
+import io.spine.protodata.ast.File
+import io.spine.protodata.ast.findOption
+import io.spine.protodata.ast.qualifiedName
+import io.spine.protodata.check
+
+/**
+ * Reports a compilation error if the [companion] option is applied
+ * to the given [field] without the [primary] option.
+ *
+ * Some options have a companion option for specifying an error message.
+ * This method ensures that a companion option is not used independently.
+ */
+internal fun checkBothApplied(
+    companion: GeneratedExtension<*, *>,
+    primary: GeneratedExtension<*, *>,
```

Comment:
Instead of having a function taking two _very_ similar arguments (in which I would have the `primary` first, BTW), I'd rather have an extension function to `GeneratedExtension`.

The reason is that  the invocations of `callFunction(someparameter, someotherparameter)` in which both of the arguments are of the same type, are prone to errors with the parameters swapped.